### PR TITLE
Fix NEC sampling if threads are involved

### PR DESCRIPTION
**Pull request by cvonelm**
ve_check_pid() only works for processes, not threads, so lo2s complained that it could not find the real NEC device id for those threads.

Rewrite the code to use ve_sysfs_path_info(). This gives us the correct sysfs path for a NEC device id, so we dont have to find it using ve_check_pid().

### CODE REVIEW

File: src/monitor/nec_monitor_main.cpp [hunk 1]
```diff
@@ -19,31 +19,30 @@
...
-    return std::optional<NecDevice>();
-}
-
 std::vector<Thread> NecMonitorMain::get_tasks_of(NecDevice device)
 {
-    std::ifstream task_stream(fmt::format("/sys/class/ve/ve{}/task_id_all", device.as_int()));
+    char sysfs_path_str[PATH_MAX];
+    memset(sysfs_path_str, 0, PATH_MAX);
+
+    ve_sysfs_path_info(device.as_int(), sysfs_path_str);
```
Comment (bmario) [inline src/monitor/nec_monitor_main.cpp:42]: return ignored.

Also, I suspect you could do the old `auto size = ve_sysfs_path_info(device.as_int(), sysfs_path_str);` to get the size of the string?

File: src/monitor/nec_monitor_main.cpp [hunk 2]
```diff
@@ -19,31 +19,30 @@
...
+    std::filesystem::path sysfs_path(sysfs_path_str);
+    sysfs_path /= "task_id_all";
```
Comment (bmario) [inline src/monitor/nec_monitor_main.cpp:44]: Technically speaking, if sysfs_path_str is allowed to be PATH_MAX, appending `/task_id_all` here produces a path potentially longer when PATH_MAX.

**Review by bmario — CHANGES_REQUESTED** [review]

File: src/monitor/nec_monitor_main.cpp (hunk 1 above)
Comment (cvonelm) [reply src/monitor/nec_monitor_main.cpp:42]: This just returns 0 on success, -1 on failure.

This is all the information that exists on that function in the world:
```
# declared in internal include file, but useful little helper function
cdef extern int ve_sysfs_path_info(int nodeid, char *ve_sysfs_path)
```

I just throw_errno() on -1, as this seems to set errno.

File: src/monitor/nec_monitor_main.cpp (hunk 2 above)
Comment (cvonelm) [reply src/monitor/nec_monitor_main.cpp:49]: I think that is something we have to live with. Hopefully nobody is putting enough NEC VE's into a single computer to outrun PATH_MAX.

**Review by bmario — APPROVED** [review]

**Pull request merged**